reportssql/dispensation_report_functions.sql

```diff
@@ -0,0 +1,147 @@
+DROP FUNCTION IF EXISTS getNumberOfTherapeuticLineARVDispensed;
+
+DELIMITER $$
+CREATE FUNCTION getNumberOfTherapeuticLineARVDispensed(
+    p_startDate DATE,
+    p_endDate DATE,
+    p_startAge INT(11),
+    p_endAge INT (11),
+    p_includeEndAge TINYINT(1),
+    p_arvName VARCHAR(255),
+    p_protocolLineNumber INT(11)) RETURNS INT(11)
+    DETERMINISTIC
+BEGIN
+    DECLARE result INT(11) DEFAULT 0;
+
+    SELECT count(DISTINCT o.order_id) INTO result
+    FROM orders o
+        JOIN drug_order do ON do.order_id = o.order_id
+        JOIN drug d ON d.drug_id = do.drug_inventory_id AND d.retired = 0
+        JOIN concept_name cn ON cn.concept_id = d.concept_id AND cn.locale = "en"
+    WHERE o.voided = 0
+        AND drugIsARV(d.concept_id)
+        AND cn.name = p_arvName
+        AND patientHasTherapeuticLine(o.patient_id, p_protocolLineNumber)
+        AND o.scheduled_date BETWEEN p_startDate AND p_endDate
+        AND drugOrderIsDispensed(o.patient_id, o.order_id)
+        AND patientAgeIsBetween(o.patient_id, p_startAge, p_endAge, p_includeEndAge);
+
+    RETURN (result);
+END$$ 
+DELIMITER ;
+
+DROP FUNCTION IF EXISTS getNumberOfAES_ARVDispensed;
+
+DELIMITER $$
+CREATE FUNCTION getNumberOfAES_ARVDispensed(
+    p_startDate DATE,
+    p_endDate DATE,
+    p_startAge INT(11),
+    p_endAge INT (11),
+    p_includeEndAge TINYINT(1),
+    p_arvName VARCHAR(255)) RETURNS INT(11)
+    DETERMINISTIC
+BEGIN
+    DECLARE result INT(11) DEFAULT 0;
+
+    SELECT count(DISTINCT o.order_id) INTO result
+    FROM orders o
+        JOIN drug_order do ON do.order_id = o.order_id
+        JOIN drug d ON d.drug_id = do.drug_inventory_id AND d.retired = 0
+        JOIN concept_name cn ON cn.concept_id = d.concept_id AND cn.locale = "en"
+    WHERE o.voided = 0
+        AND drugIsARV(d.concept_id)
+        AND cn.name = p_arvName
+        AND o.scheduled_date BETWEEN p_startDate AND p_endDate
+        AND drugOrderIsDispensed(o.patient_id, o.order_id)
+        AND patientPickedARVDrugDuringReportingPeriodWithNoTherapeuticLine(o.patient_id, p_startDate, p_endDate)
+        AND patientReasonForConsultationIsUnplannedAid(o.patient_id)
```

Comment:
@tmvumbi2 In the spec it says: If it is a PA patient then the reason for consultation must be “Unplanned Aid”. Shouldn't this check be in getNumberOfPA_ARVDispensed instead of here?